resources/views/scheduleInstrument.html

```diff
@@ -230,88 +340,336 @@ <h3>Time Scheduled</h3>
             $('#event-modal input[name="event-notes"]').val(event ? event.notes : '');
             $('#event-modal').modal();
         }
 
+        function closeEvent() {
+            removeEventLog();
+            $('#schedule-cost-error').text('');
+            $('#cost-modal').modal('hide');
+        }
+
+        function addEvent() {
+            eventToSave.id = $('#event-modal input[name="event-index"]').val();
+            eventToSave.startTime = $('#event-modal input[name="event-start-date"]').val();
+            eventToSave.endTime = $('#event-modal input[name="event-end-date"]').val();
+            eventToSave.project = project;
+            eventToSave.instrument = instrument ? instrument : $('#instrumentDropDown select[name="instrumentDropDown"]').val();
+            eventToSave.name = $('#event-modal input[name="event-name"]').val();
+            eventToSave.notes = $('#event-modal input[name="event-notes"]').val();
+            eventToSave.instrumentoperator = $('#instrumentOperatorDropDown select[name="instrumentOperatorDropDown"]').val();
+
+            if (eventToSave.endTime < eventToSave.startTime) {
+                alert('End date must be after start date');
+                return;
+            }
+
+            const events = calendar.getEvents();
+            // validate that the events time do not overlap with existing events
+            for (let i = 0; i < events.length; i++) {
+                if (eventToSave.id != events[i].id) {
+                    let newEvent = {
+                        start: new Date(eventToSave.startTime),
+                        end: new Date(eventToSave.endTime)
+                    };
+                    let existingEvent = {
+                        start: events[i].start,
+                        end: events[i].end
+                    };
+
+                    if (eventsOverlap(newEvent, existingEvent)) {
+                        alert('New schedule overlaps with existing instrument schedule');
+                        return;
+                    }
+                }
+            }
+
+            // get the paymentMethod and percentPayment from the html fields
+            paymentMethods = [];
+            let paymentMethodDropDowns = document.querySelectorAll('#paymentMethodDropDown select');
+            let percentPaymentFields = document.querySelectorAll('#paymentMethodPercent');
+            for (let i = 0; i < paymentMethodDropDowns.length; i++) {
+                let paymentMethodId = paymentMethodDropDowns[i].value;
+                let paymentMethodPercent = percentPaymentFields[i].value;
+                paymentMethods.push({
+                    paymentMethod: paymentMethodId,
+                    percentPayment: paymentMethodPercent
+                });
+            }
+
+            fetchInstrumentCosts(eventToSave.instrument, eventToSave.startTime, eventToSave.endTime, true);
+
+            // TODO: Make sure user has not exceeded instrument time quota
```

Comment:
Important TODOs but when they're implemented let's be sure to enforce them on the server and not in the client.